embassy-rp/src/adc.rs

```diff
@@ -176,19 +198,71 @@ impl<'d> Adc<'d, Async> {
         }
     }
 
-    pub async fn read(&mut self, pin: &mut Pin<'_>) -> Result<u16, Error> {
-        Self::sample_async(pin.channel()).await
-    }
-
-    pub async fn read_temperature(&mut self) -> Result<u16, Error> {
+    pub async fn read_many<S: AdcSample>(
+        &mut self,
+        ch: &mut Channel<'_>,
+        buf: &mut [S],
+        dma: impl Peripheral<P = impl dma::Channel>,
+    ) -> S::Result {
```

Comment:
same as in #1715 , changing the return type like this is very surprising/unconventional. Can you change this to always return `Result<(), Error>`?

Reply:
> the possible overhead of unwrapping for an error

that overhead is tiny, and the compiler will probably optimize it out anyway.

> we'd prefer to split the method in two

that sounds good too!